### PR TITLE
Bug fix for lens_plot.py update

### PR DESCRIPTION
**Pull request by jocelyn-mcmahon-16**
Fixed error regarding indices, updated labeling defaults.

### CODE REVIEW

**Comment by sibirrer** [comment]
@jocelyn-mcmahon-16 there are some tests failing. You can execute the tests locally with e.g. $py.test test/test_Plots

**Comment by sibirrer** [comment]
@jocelyn-mcmahon-16 just let @ajshajib know when it's ready to be reviewed again. Thank you!

**Comment by jocelyn-mcmahon-16** [comment]
Thanks Simon!
@ajshajib  I think this is ready to be reviewed again. Thanks!

File: lenstronomy/Plots/lens_plot.py [hunk 1]
```diff
@@ -292,25 +295,27 @@ def point_source_plot(
...
+    param color_list: list of strings to represent the color for each source's images. Default None.
     :param kwargs: additional plotting keyword arguments
     :return: matplotlib axis instance with figure
     """
     from lenstronomy.LensModel.Solver.lens_equation_solver import LensEquationSolver
 
+    name_List_ = []
     if name_list is None and index is None:
-        name_list = _NAME_LIST
+        name_List_ = _NAME_LIST
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:308]:
```suggestion
        name_list_ = _NAME_LIST
```

File: lenstronomy/Plots/lens_plot.py [hunk 2]
```diff
@@ -292,25 +295,27 @@ def point_source_plot(
     :param source_x: x-position of source
     :param source_y: y-position of source
     :param name_list: list of names of images
-    :param name_list: list of strings, longer or equal the number of point sources. If changing this parameter, input as name_list=[...]
+    :param name_list: list of strings, longer or equal the number of point sources. If changing this parameter, input as name_list=[[...], [...]]
     :param index: number of sources, an integer number. Default None.
+    param color_list: list of strings to represent the color for each source's images. Default None.
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:300]:
```suggestion
    :param color_list: list of strings to represent the color for each source's images. Default "k".
```

File: lenstronomy/Plots/lens_plot.py [hunk 3]
```diff
@@ -337,19 +342,24 @@ def point_source_plot(
...
+            alpha=0.5,
         )
-        ax.text(x_, y_, name_list[i], fontsize=20, color="k")
+        ax.text(x_, y_, name_List_[i], fontsize=20, color=color)
     x_source, y_source = pixel_grid.map_coord2pix(source_x, source_y)
     ax.plot(
         x_source * delta_pix_x + origin[0],
         y_source * delta_pix + origin[1],
-        "*k",
+        color,
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:362]: Here, `ax.plot()` expects a single color, I think, not a list. But, the documentation implies that a list of color strings is being passed.

File: lenstronomy/Plots/lens_plot.py [hunk 4]
```diff
@@ -41,6 +41,7 @@ def lens_model_plot(
     fast_caustic=True,
     name_list=None,
     index=None,
+    color_list="k",
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:44]: Corresponding line needs to be added in the docstring.

File: test/test_Plots/test_lens_plot.py [hunk 5]
```diff
@@ -127,8 +170,9 @@ def test_lens_model_plot(self):
                 deltaPix=0.5,
                 sourcePos_x=x_sources[i],
                 sourcePos_y=y_sources[i],
-                name_list=["Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"],
+                name_list=name_List[i],
                 index=i,
+                color_list=color_List[i],
```
Comment (ajshajib) [inline test/test_Plots/test_lens_plot.py:175]:
```suggestion
                color_list=color_list[i],
```

File: test/test_Plots/test_lens_plot.py [hunk 6]
```diff
@@ -119,6 +161,7 @@ def test_lens_model_plot(self):
         plt.close
 
         for i in range(len(x_sources)):
+            color_List = ["k", "b", "g", "r", "c", "m", "y"]
```
Comment (ajshajib) [inline test/test_Plots/test_lens_plot.py:164]:
```suggestion
            color_list = ["k", "b", "g", "r", "c", "m", "y"]
```

File: test/test_Plots/test_lens_plot.py [hunk 7]
```diff
@@ -67,6 +67,10 @@ def test_lens_model_plot(self):
         x_source2, y_source2 = -0.15, -0.12
         x_sources = [x_source, x_source2]
         y_sources = [y_source, y_source2]
+        name_List = [
```
Comment (ajshajib) [inline test/test_Plots/test_lens_plot.py:70]:
```suggestion
        name_lists = [
```

File: test/test_Plots/test_lens_plot.py [hunk 8]
```diff
@@ -110,7 +116,43 @@ def test_lens_model_plot(self):
                 deltaPix=0.5,
                 sourcePos_x=x_sources[i],
                 sourcePos_y=y_sources[i],
-                name_list=["Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"],
+                name_list=name_List[i],
```
Comment (ajshajib) [inline test/test_Plots/test_lens_plot.py:119]:
```suggestion
                name_list=name_lists[i],
```

File: lenstronomy/Plots/lens_plot.py [hunk 9]
```diff
@@ -337,19 +342,24 @@ def point_source_plot(
...
         ax.plot(
-            x_, y_, "dk", markersize=4 * (1 + np.log(np.abs(mag_images[i]))), alpha=0.5
+            x_,
+            y_,
+            str("d" + color),
+            markersize=4 * (1 + np.log(np.abs(mag_images[i]))),
+            alpha=0.5,
         )
-        ax.text(x_, y_, name_list[i], fontsize=20, color="k")
+        ax.text(x_, y_, name_List_[i], fontsize=20, color=color)
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:357]: I think you can't pass a list of colors here.

File: lenstronomy/Plots/lens_plot.py [hunk 10]
```diff
@@ -292,25 +295,27 @@ def point_source_plot(
...
 
+    name_List_ = []
     if name_list is None and index is None:
-        name_list = _NAME_LIST
+        name_List_ = _NAME_LIST
     elif name_list is None and index is not None:
         name_list = _NAME_LIST
         for i in range(len(name_list)):
-            name_list[i] = str(index + 1) + name_list[i]
+            name_List_.append(str(index + 1) + name_list[i])
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:312]:
```suggestion
            name_list_.append(str(index + 1) + name_list[i])
```

File: lenstronomy/Plots/lens_plot.py [hunk 11]
```diff
@@ -292,25 +295,27 @@ def point_source_plot(
...
     elif name_list is not None and index is None:
-        name_list = name_list
+        name_List_ = name_list
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:314]:
```suggestion
        name_list_ = name_list
```

File: lenstronomy/Plots/lens_plot.py (hunk 4 above)
Comment (ajshajib) [reply lenstronomy/Plots/lens_plot.py:44]: If this is a list of colors, the default should be a list, e.g., `["k"]`.

File: lenstronomy/Plots/lens_plot.py [hunk 12]
```diff
@@ -278,6 +280,7 @@ def point_source_plot(
     source_y,
     name_list=None,
     index=None,
+    color_list="k",
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:283]: The default value should be a list.

**Review by ajshajib — CHANGES_REQUESTED** [review]
Thanks a lot for the updates! I have made a few suggestions, many of them are already as suggested commits. One way to go forward with implementing the suggestions would be clicking on "Commit suggestion" if you would like to accept the suggestion, then pulling from the remote branch to your local, and then working on your local copy to implement the rest of the suggestions before pushing your local commits to the remote again.

**Comment by ajshajib** [comment]
Hi @jocelyn-mcmahon-16, thanks a lot for the updates. When it's ready for review again, please click on the "Re-request review" button at the top right of this page in the "Reviewers" section.

File: lenstronomy/Plots/lens_plot.py [hunk 13]
```diff
@@ -337,19 +342,24 @@ def point_source_plot(
     mag_images = lens_model.magnification(theta_x, theta_y, kwargs_lens)
 
     x_image, y_image = pixel_grid.map_coord2pix(theta_x, theta_y)
+    color = color_value
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:345]:
```suggestion
```

File: lenstronomy/Plots/lens_plot.py [hunk 14]
```diff
@@ -337,19 +342,24 @@ def point_source_plot(
...
         ax.plot(
-            x_, y_, "dk", markersize=4 * (1 + np.log(np.abs(mag_images[i]))), alpha=0.5
+            x_,
+            y_,
+            str("d" + color),
+            markersize=4 * (1 + np.log(np.abs(mag_images[i]))),
+            alpha=0.5,
         )
-        ax.text(x_, y_, name_list[i], fontsize=20, color="k")
+        ax.text(x_, y_, name_list_[i], fontsize=20, color=color)
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:357]:
```suggestion
        ax.text(x_, y_, name_list_[i], fontsize=20, color=color_value)
```

File: lenstronomy/Plots/lens_plot.py [hunk 15]
```diff
@@ -337,19 +342,24 @@ def point_source_plot(
...
     x_source, y_source = pixel_grid.map_coord2pix(source_x, source_y)
     ax.plot(
         x_source * delta_pix_x + origin[0],
         y_source * delta_pix + origin[1],
-        "*k",
+        color,
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:362]:
```suggestion
        color_value,
```

File: lenstronomy/Plots/lens_plot.py [hunk 16]
```diff
@@ -278,6 +280,7 @@ def point_source_plot(
     source_y,
     name_list=None,
     index=None,
+    color_value="k",
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:283]:
```suggestion
    color="k",
```

**Review by ajshajib — APPROVED** [review]
Thanks a lot, @jocelyn-mcmahon-16! I am making some minor changes and will merge the PR after the tests pass.

File: lenstronomy/Plots/lens_plot.py [hunk 17]
```diff
@@ -123,6 +124,7 @@ def lens_model_plot(
             source_y=sourcePos_y,
             name_list=name_list,
             index=index,
+            color_value=color_value,
```
Comment (ajshajib) [inline lenstronomy/Plots/lens_plot.py:127]:
```suggestion
            color=color_value,
```

**Pull request merged**